### DISCUSSION TITLE
Option to set all months to 30 days for timetable sync

### DISCUSSION THREAD

**Discussion by felixprigge** [OpenTTD/OpenTTD#12630] [Features]
Hi, if the game had a setting to set all months to a length of 30 days, timetable synchonisation between different lines would become much easier. Constructing the timetable length to a multiple of 30 days would result in static arrival and departure days (day number) each month. Different train lines could then be synchronized more easily. Would only have a small impact on economy. Loss of reality is also small. What do you think?

**Comment by 2TallTyler** [comment]
In 14.0, you can set the timetable units to Seconds, which shows relative seconds to the current moment. To synchronize multiple lines, you could build a "clock" train that has a round trip duration matching your chosen frequency -- say 60 seconds -- and set against the departure time of its first order.

**Reply by felixprigge** [reply]
Hi 2TallTyler, thanks for your quick reply. The new timetable unit seems to be a nice feature but I think having static arrival and departure "moments" in the timetable is more clearly. 
I maybe try to change the timer_game_common.cpp for myself and test how it turns out. Are there any events happening exactly on 31sts that can cause problems? Like introducing new vehicles? I could also just extend all months to 31 days to avoid problems, I guess :) 
I will share my results

**Reply by 2TallTyler** [reply]
30-day months are no problem, but you will encounter some strangeness with 360-day years. 😉 The economy should keep working, but the date display will be a bit weird due to conversion errors between `Date` (days since January 1, year 0) and `YearMonthDay`.

A better option might be to add static moments for Seconds mode. Timetables are based on game ticks so the units only exist in the user interface. I created Seconds mode using relative offsets because static moments require the player to know the current moment, and we only show the calendar date. At slow speeds this is imprecise, and if frozen the date is meaningless.

That said, when the calendar is frozen the player doesn't need to know the date at all, so we can repurpose that display in the status bar for other purposes. I'd suggest an optional setting to instead show the elapsed playtime in `D Days HH:MM:SS`. If you feel up for coding that, I'd be happy to review a PR and help get it merged. 😃

**Reply by felixprigge** [reply]
I did a quick and dirty implementation of the 30-day months and it works well with my current savegame. Just experienced a small jump into the future, which is fine for me, as I play with old trains anyway. But I see that an implementation where people can choose between realistic and 30-day months is probably not worth it. Especially now with the wallclock mode. I guess the people aiming at creating nice timetables will mostly play in that mode. I am pretty sure that I will switch to wallclock with frozen year for my next game as well :) Thank you for putting so much time and effort in it.

That being said, your suggestion to implement static moments in wallclock mode sounds interesting. I dont know if I have the skills for such an implementation, but I will have a deeper look at it. Displaying the elapsed time in the status bar instead of date only works if the calendar is frozen. But the static moments should be available also if calendar is not frozen. Maybe both could be displayed. Having elapsed play time displayed is probably a nice feature in every game mode anyway. Lets see :)
